### ISSUE TITLE
[Rooms list] Do not suggest collapse the unique section

### ISSUE THREAD

**Issue by giomfo** [element-hq/element-android#5347]
#### What would you like to do?
In the rooms list, the sections are collapsable
<img width="93" alt="image" src="https://user-images.githubusercontent.com/8969772/155544714-c51efb61-2a68-4453-8506-32539f2cd34d.png">

This is ok when there are severals section, but irrelevant when there is only one section.
I suggest here disabling and hiding the option when there is only one section (this is the case for example when I don't have pending invites and when I didn't taf room (favorite/low priority))
When we select a space the case where there is only one section is more frequent

**Comment by giomfo** [comment]
We can prepare a PR if you agree with the suggestion

**Comment by daniellekirkwood** [comment]
Hi @giomfo 

Please raise the PR. 

If there is only one section, then that section should not be collapsible. We recently made this same decision and change on iOS too.

Thanks

**Comment by bmarty** [comment]
I do not like the idea behind this issue TBH. It should still be collapsible, just for the fact that user can quickly see what's below (= nothing) without having to scroll to the bottom.

**Comment by ganfra** [comment]
Also the PR #5580 had issues, so reverted for now.

**Comment by bmarty** [comment]
OK, so, we will have to ensure that this is working:

- an invite is there, so the sections can be collaped
- the user collapse the "rooms" section
- the invite is gone (either by accepting or rejecting it)

-> make sure that the "rooms" section is expanded, or expandable again

**Comment by bmarty** [comment]
(this is not the issue we have seen though)

The issues was:
- even if the arrow is hidden, the header is still clickable and the list collapse
- sometimes the arrows are not displayed, even if there are several sections

All the fixes are in #5616 from @Florian14 , but we decided to revert the original PR also to discuss a bit more about this "feature"

**Comment by Florian14** [comment]
> OK, so, we will have to ensure that this is working:
> 
>     * an invite is there, so the sections can be collaped
> 
>     * the user collapse the "rooms" section
> 
>     * the invite is gone (either by accepting or rejecting it)
> 
> 
> -> make sure that the "rooms" section is expanded, or expandable again

I don't think so this case was handled before but it should be fine with #5616: a section is expanded according to a flag value **or** if there is no other section

**Comment by bmarty** [comment]
There is also another pb (which is maybe not new)

- on "Direct Message" tab collapse a section
- go to "Rooms" tab
- go back to "Direct Message" tab

-> The section is expanded, but the state is not updated, it means that the user has to click twice on the section header to collapse it.

**Comment by giomfo** [comment]
@Florian14 can we close this issue?
